packages/dashboard/src/components/widgets/primitives/input/ConfirmationModal.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { Dispatch, useEffect, useState } from 'react';
+import useUpdateAssetPropertyValue from '~/components/widgets/primitives/input/hooks/useUpdateAssetPropertyValue';
+import useDescribeAssetProperty from '~/components/widgets/primitives/input/hooks/useDescribeAssetProperty';
+import { ErrorDetails } from '@aws-sdk/client-iotsitewise';
+import { NonCancelableEventHandler } from '@cloudscape-design/components/internal/events';
+import { Box, Button, ButtonProps, Modal } from '@cloudscape-design/components';
+
+const ConfirmationModal = ({
+  showConfirmationModal,
+  setShowConfirmationModal,
+  submitLabel,
+  input,
+  setErrorMessage,
+  assetId,
+  propertyId,
+  setSuccessMessage,
+  successMessage,
+  setIsSending,
+}: {
+  showConfirmationModal: boolean;
+  setShowConfirmationModal: Dispatch<boolean>;
+  submitLabel: string;
+  input: string;
+  setErrorMessage: Dispatch<string | null>;
+  assetId: string;
+  propertyId: string;
+  setSuccessMessage: Dispatch<string | null>;
+  successMessage: string | null;
+  setIsSending: Dispatch<boolean>;
+}) => {
+  const [propertyName, setPropertyName] = useState<string>();
+  const updateAssetPropertyValue = useUpdateAssetPropertyValue();
+  const describeAssetProperty = useDescribeAssetProperty();
+
+  useEffect(() => {
+    (async () => {
+      if (assetId && propertyId) {
+        try {
+          const property = await describeAssetProperty({ assetId, propertyId });
+          const name = property?.assetProperty?.name;
+          if (name) {
+            setPropertyName(name);
+          }
+        } catch (e) {
+          setErrorMessage((e as ErrorDetails).message ?? null);
```

Comment:
blocking: when error isn't as expected, it will set as having no error. should fall back to an error message such as 'unknown error'